posts/views.py

```diff
@@ -5,19 +5,22 @@
 from django.views.generic import ListView, CreateView, UpdateView, DeleteView
 from django.urls import reverse_lazy
 from .forms import PostForm
+from django.contrib.auth.mixins import LoginRequiredMixin
 
 
 # Create your views here.
-class PostListView(ListView): 
+class PostListView(LoginRequiredMixin, ListView): 
     model = Post
     template_name = 'main.html'
     context_object_name = 'posts'
+    login_url = reverse_lazy('login')
 
     def get_queryset(self):
         base_query = super().get_queryset()
         data = base_query.order_by('-created_date', '-created_time')
         return data
-     
+    
+      
 class PostEditView(UpdateView):
```

Comment:
Why we didn't put this "LoginRequiredMixin" here?

Reply:
i dont know, do you want me to open ticket and inlude it in all of them ?

Reply:
added in all of them